hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -133,6 +133,14 @@ finalizeTx TinyWallet{sign, coverFee} ctx ChainStateAt{chainState} partialTx = d
       throwIO (NoFuelUTXOFound :: PostTxError Tx)
     Left ErrNotEnoughFunds{} ->
       throwIO (NotEnoughFuel :: PostTxError Tx)
+    Left ErrScriptExecutionFailed{scriptFailure} ->
+      throwIO
+        ( PlutusValidationFailed
+            { plutusFailure = show $ snd scriptFailure
+            , plutusDebugInfo = ""
+            } ::
+            PostTxError Tx
```

Comment:
I realize the `scriptFailure :: TransactionScriptFailure c` from the ledger is more broad than our `PlutusValidationFailed` here. The corresponding `cardano-api` type would be `ScriptExecutionError`.

So maybe we should have another `PostTxError` constructor for this generic script execution error using `ScriptExecutionError` (and change the other source for `PlutusValidationFailed` into the `ScriptErrorEvaluationFailed` data constructor)

Reply:
does this request invalidate the [above](https://github.com/input-output-hk/hydra/pull/680#discussion_r1093399756)?

Reply:
Yes, it's a new opinion compared to the quick draft with the TODOs I did in the ensemble. Is the explanation not clear in my comment above? I'm afraid I can't write it more clear.

Reply:
I'm sorry but the explanation is not really clear to me.

Are you saying we should be using `evaluateTransactionExecutionUnits` from `cardano-api` instead from the `ledger` in `coverFee_` so we get more clear/specific error?

Reply:
should we tackle this improvement as part of this PR?